src/components/comment/comment.ts

```diff
@@ -0,0 +1,41 @@
+export const initComments = () => {
+  const commentsWithRepliesEnabled = document.querySelectorAll(
+    '.js-comment-with-replies-enabled'
+  );
+
+  for (const comment of [...commentsWithRepliesEnabled]) {
+    const replyButton = comment.querySelector('.js-comment-reply-button');
+    const replyForm = comment.querySelector('.js-comment-reply-form');
+    // The cancel button is inside the Comment Form, so we need to reach inside
+    // that component to grab the button. This feels a little awkward since
+    // we're directly manipulating a child component, but this is a lot simpler
+    // than other potential solutions. Other options included:
+    // - Having the Comment Form emit a `cancel` event
+    // - Having the Comment Form have a block that you could pass a button into
+    const cancelButton = comment.querySelector('.js-cancel-reply');
+
+    replyButton?.addEventListener('click', () => {
+      comment.classList.add('is-replying');
+      replyButton.setAttribute('hidden', '');
+      replyForm?.removeAttribute('hidden');
+      const firstInput = replyForm?.querySelector(
+        'textarea, input'
+      ) as HTMLElement | null;
+
+      // Without this Timeout, VoiceOver does not properly focus the first input
+      // (though it works outside of VoiceOver).
+      // I went back and forth on whether or not to force this behavior in
+      // VoiceOver, since it will skip them past context around the comment
+      // (e.g. you may use simple HTML or markdown).
+      // I ended up adding this timeout to force the focus so that we provide
+      // an equivalent experience for all users.
+      setTimeout(() => firstInput?.focus(), 0);
```

Comment:
I'm conflicted on this. Thoughts?

Reply:
@Paul-Hebert Can you elaborate? You are conflicted on using a `setTimeout()`? Or something else?

Reply:
Sure! 

The initial issue (#1389) specifically mentioned the first input should gain focus when the form is shown. I think this is a great UX for most people since it allows them to skip a click and start typing right away.

To get this working in VoiceOver I had to add the timeout. As I was doing this, I started to question whether this was a good experience for screen reader users or not. If they're not able to visually see the page, the text in between the reply button and the first textarea will not be read:

<img width="933" alt="Screen Shot 2021-07-20 at 4 12 37 PM" src="https://user-images.githubusercontent.com/5798536/126406699-7ab7e880-404e-41f5-bcc1-b55404d862d4.png">

It seemed weird to not have this feature work the same for screen reader users and folks not using screen readers, but it also seemed weird to add in a `timeout` hack to enable an experience for screen reader users which is arguably worse.

Does that context help?

Reply:
Thanks for the extra context, @Paul-Hebert! 

> As I was doing this, I started to question whether this was a good experience for screen reader users or not. If they're not able to visually see the page, the text in between the reply button and the first textarea will not be read

Good point. I'm wondering if this is a case where we might consider setting the focus on the heading (would require adding a `z-index`)? Or maybe wrapping the heading + instructions in an ARIA Live region and still have the focus jump to the first input? 🤔

Reply:
Yeah, good question 🤔 

Those both feel a little weird to me but I can't put my finger on why 🤔 . That said, the current experience also feels a little weird in a screen reader

I can try them out and see how they feel. :+1:

Reply:
> Those both feel a little weird to me but I can't put my finger on why 🤔 . That said, the current experience also feels a little weird in a screen reader

@Paul-Hebert I hear ya'. This is a very interesting UX to figure out. If you wanna jump on a call to share what each experience feels like, I'm available. 😉

Reply:
Yeah that sounds good. I'm going to respond to the other pieces of feedback first and then circle back to this.

Reply:
Depending on timing, maybe we could walk through this as a group in the design team meeting?

Reply:
Oh, wait! What if I used `aria-describedby` on the help text. Then it might be read aloud... 🤔

Reply:
Yep. `aria-describedby` is the way to go :+1: 

I'll share more info in the design meeting.

Reply:
Yay, great solution, @Paul-Hebert! 🎉